targetcli/ui_backstore.py

```diff
@@ -611,9 +611,22 @@ def ui_command_create(self, name, size, cfgstring, wwn=None,
 
         try:
             so = UserBackedStorageObject(name, size=size, config=config,
-                                         wwn=wwn, hw_max_sectors=hw_max_sectors)
-        except:
-            raise ExecutionError("UserBackedStorageObject creation failed.")
+                                         wwn=wwn, hw_max_sectors=hw_max_sectors,
+                                         max_data_area_mb=max_data_area_mb)
+        except TypeError as e:
+            if "max_data_area_mb" not in str(e):
+                raise ExecutionError("UserBackedStorageObject creation failed.")
+
+            if max_data_area_mb is not None:
+                raise ExecutionError("UserBackedStorageObject creation failed."
+                                     " max_data_area_mb not supported.")
+
+            try:
+                so = UserBackedStorageObject(name, size=size, config=config,
+                                             wwn=wwn,
+                                             hw_max_sectors=hw_max_sectors)
+            except:
+                raise ExecutionError("UserBackedStorageObject creation failed.")
```

Comment:
Andy,

One question/comment for my patch. I was not sure how much rtslib compat support was needed. Do you normally require users to update rtslib and targetcli together? The code above handles where you have a new targetcli but old rtslib that does not support max_data_area support. I can drop this if you want.

Reply:
Going forward then to make it easier, do you think it would be ok to separate the write(enable=1) step from the constructor? So for tcmu you might do:
'
dev = UserBackedStorageObject(pass_in_required_args_like_name_and_config_like_usual.
                                                       new_arg_deferred_enable=true)
try:
   dev.set_control("setting", value)
catch RTSLibControlError:
    # app decides if fatal or not

dev.enable = 1
'

We won't have to update rtslib for every new control setting. rtslib could also have a function that reads the info file and returns a list of control settings. In this deferred mode, targetcli could then display and set them  similar to attributes. We would then not have to update targetcli for every new setting too. So you would do

/backstores/user:rbd>  create name=test cfgstring=rbd/test

// isimiar to get attribute
/backstores/user:rbd/test> get control
dev_config
dev_size
hw_block_size
max_data_area_mb

/backstores/user:rbd/test> set control max_data_area_mb=10

/backstores/user:rbd/test> enable

Reply:
This sounds like a solid improvement to me.

Reply:
Ok. I am going to close this and the rtslib PR.

Will open new PRs when I have code for this.

Thanks.